tree/ntuple/v7/src/RField.cxx

```diff
@@ -785,7 +787,14 @@ ROOT::Experimental::RClassField::RClassField(std::string_view fieldName, std::st
 	     RSubFieldInfo{kBaseClass, static_cast<std::size_t>(baseClass->GetDelta())});
       i++;
    }
+   std::unordered_set<void *> realMembers;
+   for (auto rm : ROOT::Detail::TRangeStaticCast<TRealData>(*fClass->GetListOfRealData()))
+      realMembers.insert(rm->GetDataMember());
    for (auto dataMember : ROOT::Detail::TRangeStaticCast<TDataMember>(*fClass->GetListOfDataMembers())) {
+      // Skip, for instance, unscoped enum constants defined in the class
+      if (realMembers.count(dataMember) == 0)
+         continue;
```

Comment:
I think that, instead of iterating twice over the list of data members, perhaps we can use `TDataMember::IsEnum()` in combination with either `GetTypeName()` (which should contain `something::(anonymous)`) or `GetOffset()` (which for enumerators of an unscoped enum that appear in the parent scope, returns strangely high values), e.g. for `StructWithEnums`:
```
a: IsEnum=0 type=int offset @4
b: IsEnum=0 type=int offset @8
A1: IsEnum=1 type=StructWithEnums::(anonymous) offset @94646202990808
A2: IsEnum=1 type=StructWithEnums::(anonymous) offset @94646202990896
E1: IsEnum=1 type=StructWithEnums::DeclE offset @94646202991224
E2: IsEnum=1 type=StructWithEnums::DeclE offset @94646202991312
E42: IsEnum=1 type=StructWithEnums::DeclE offset @94646202991464
```

Reply:
The problem here is that also real data members that are declared as enums have `IsEnum() == true`. While at the moment we don't support enums, at some point we want to.  And then we need to distinguish between `enum { E1 };` and `E1 e1;`.

Reply:
The definitive comprehensive answer to all your question is in: `TStreamerInfo::Build` :)

The short answer to skip enum declaration is:
```
      if (dataMember->Property() & kIsUnionMember) {
         continue;
      }
```

Reply:
Are you sure that you don't mean `kIsStatic`?

Reply:
In `TStreamerInfo::Build`, `static` is indirectly taken care  of by:
```
      TMemberStreamer* streamer = 0;
      Int_t offset = GetDataMemberOffset(dm, streamer);
      if (offset == kMissing) {
         continue;
      }
```
where the static data member have the offset `kMissing` (i.e. -1)

Reply:
In `TStreamerInfo::Build`, `static` is indirectly taken care  of by:
```
      TMemberStreamer* streamer = 0;
      Int_t offset = GetDataMemberOffset(dm, streamer);
      if (offset == kMissing) {
         continue;
      }
```
where the static data member have the return value of `TStreamerInfo::GetDataMemberOffset` is  `kMissing`

Reply:
This is obviously sub-optiomal and we should consider updating `TDataMember` to mark the `static` and `union` members are transient.

Reply:
> Are you sure that you don't mean kIsStatic?

And actually @jblomer, you are completely right ... I mis-read the code last night (`Union` vs `Enum`).